core/chain-configs/src/test_genesis.rs

```diff
@@ -94,175 +103,154 @@ struct UserAccount {
-        self.minimum_validators_per_shard = Some(minimum_validators_per_shard);
+        self.minimum_validators_per_shard = minimum_validators_per_shard;
         self
     }
 
     pub fn target_validator_mandates_per_shard(
         mut self,
         target_validator_mandates_per_shard: NumSeats,
     ) -> Self {
-        self.target_validator_mandates_per_shard = Some(target_validator_mandates_per_shard);
+        self.target_validator_mandates_per_shard = target_validator_mandates_per_shard;
         self
     }
 
-    pub fn shuffle_shard_assignment_for_chunk_producers(mut self, shuffle: bool) -> Self {
-        self.shuffle_shard_assignment_for_chunk_producers = Some(shuffle);
+    pub fn shuffle_shard_assignment_for_chunk_producers(
+        mut self,
+        shuffle_shard_assignment_for_chunk_producers: bool,
+    ) -> Self {
+        self.shuffle_shard_assignment_for_chunk_producers =
+            shuffle_shard_assignment_for_chunk_producers;
         self
     }
 
     // Validators with performance below 80% are kicked out, similarly to
     // mainnet as of 28 Jun 2024.
     pub fn kickouts_standard_80_percent(mut self) -> Self {
-        self.block_producer_kickout_threshold = Some(80);
-        self.chunk_producer_kickout_threshold = Some(80);
-        self.chunk_validator_only_kickout_threshold = Some(80);
+        self.block_producer_kickout_threshold = 80;
+        self.chunk_producer_kickout_threshold = 80;
+        self.chunk_validator_only_kickout_threshold = 80;
         self
     }
 
     // Only chunk validator-only nodes can be kicked out.
     pub fn kickouts_for_chunk_validators_only(mut self) -> Self {
-        self.block_producer_kickout_threshold = Some(0);
-        self.chunk_producer_kickout_threshold = Some(0);
-        self.chunk_validator_only_kickout_threshold = Some(50);
+        self.block_producer_kickout_threshold = 0;
+        self.chunk_producer_kickout_threshold = 0;
+        self.chunk_validator_only_kickout_threshold = 50;
         self
     }
 
     pub fn build(self) -> EpochConfig {
-        let epoch_length = self.epoch_length.unwrap_or_else(|| {
-            let default = 5;
-            tracing::warn!(
-                "Epoch config epoch_length not explicitly set, defaulting to {:?}.",
-                default
-            );
-            default
-        });
-        let shard_layout = self.shard_layout.unwrap_or_else(|| {
-            let default = ShardLayout::single_shard();
-            tracing::warn!(
-                "Epoch config shard_layout not explicitly set, defaulting to {:?}.",
-                default
-            );
-            default
-        });
-        let validators_spec = self.validators_spec.unwrap_or_else(|| {
-            let default = ValidatorsSpec::DesiredRoles {
-                block_and_chunk_producers: vec!["validator0".to_string()],
-                chunk_validators_only: vec![],
+        let defaults = Self::default();
```

Comment:
So this warns even if the value were explicitly set but to the default value.

How about just print the whole config out so it can be inspected if needed?

Reply:
> How about just print the whole config out so it can be inspected if needed?

https://github.com/near/nearcore/pull/12554#discussion_r1876249591

Will do. That's exactly what I had in mind :))